### PR TITLE
fix decode in encoder decoder

### PR DESCRIPTION
**Pull request by DeNeutoy**

### CODE REVIEW

**Comment by DeNeutoy** [comment]
@maxdel This will likely be relevant for you too if you are using the encoder decoder stuff.

File: allennlp/models/encoder_decoders/simple_seq2seq.py [hunk 1]
```diff
@@ -280,8 +280,6 @@ def decode(self, output_dict: Dict[str, torch.Tensor]) -> Dict[str, torch.Tensor
             predicted_tokens = [self.vocab.get_token_from_index(x, namespace="target_tokens")
                                 for x in indices]
             all_predicted_tokens.append(predicted_tokens)
-        if len(all_predicted_tokens) == 1:
```
Comment (matt-gardner) [inline allennlp/models/encoder_decoders/simple_seq2seq.py]: This is removing the batch dimension, which we don't do here anymore?  Is that the issue?

Comment (DeNeutoy) [reply allennlp/models/encoder_decoders/simple_seq2seq.py]: Yep, exactly. Doing this results in the decoded output being the first word if there is a single element in the batch.

Comment (maxdel) [reply allennlp/models/encoder_decoders/simple_seq2seq.py]: Exactly, I also fixed this issue in my local version.

**Pull request merged**